src/v/rpc/connection_cache.cc

```diff
@@ -100,20 +100,53 @@ ss::future<> connection_cache::stop() {
 }
 
 ss::shard_id connection_cache::shard_for(
-  model::node_id self,
+  model::node_id src_node,
   ss::shard_id src_shard,
-  model::node_id n,
+  model::node_id dest_node,
   ss::shard_id total_shards) const {
     if (total_shards <= _max_connections) {
         return src_shard;
     }
 
```

Comment:
Having an overall explanation with a small ascii chart (maybe use a line instead of a ring) would be great here I think